src/pets/components/PetForm.tsx

```diff
@@ -0,0 +1,213 @@
+"use client";
+
+import { useForm } from "react-hook-form";
+import { useRouter } from "next/navigation";
+import { createPetAction } from "@/pets/actions/pets-actions";
+import { PawPrint, Calendar } from "lucide-react";
+import { useState, useEffect } from "react";
+
+type PetFormInputs = {
+  name: string;
+  species: string;
+  breed?: string;
+  color?: string;
+  sex?: "MALE" | "FEMALE" | "UNKNOWN";
+  dateOfBirth?: string;
+  castrated?: boolean;
+  urlPhoto?: string;
+  registrationDate: string;
+};
+
+export default function PetForm() {
+  const router = useRouter();
+  const [errorMessage, setErrorMessage] = useState("");
+  const [speciesOptions, setSpeciesOptions] = useState<
+    { value: string; label: string }[]
+  >([]);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm<PetFormInputs>();
+
+  useEffect(() => {
+    async function fetchSpecies() {
+      try {
+        const res = await fetch("/api/pets/species");
```

Comment:
fetching data is not a presentation concern, how about moving this a repository layer